servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -307,8 +308,21 @@ void SceneShaderForwardClustered::ShaderData::set_code(const String &p_code) {
 							continue;
 						}
 
+						RD::PipelineDepthStencilState depth_stencil;
+						if (depth_test != DEPTH_TEST_DISABLED) {
+							if (depth_pre_pass_enabled) {
```

Comment:
We need a finer grain check here, as this only works for fragments that write to the depth buffer, it shouldn't be used for regular transparent objects as it will result in them not getting drawn at all.

Reply:
Ah, true. I thought true alpha was handled by a different pipeline version. Will change that to only trigger when something that has depth is rendered.